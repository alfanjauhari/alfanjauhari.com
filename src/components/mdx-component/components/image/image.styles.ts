import { styled } from '@/theme';
import Image from 'next/image';

export const StyledImage = styled(Image, {
  my: '$5',
  objectFit: 'cover',
});
