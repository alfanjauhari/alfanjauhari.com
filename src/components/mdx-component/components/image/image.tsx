import { ComponentProps } from 'react';
import { StyledImage } from './image.styles';

export type ImageProps = ComponentProps<typeof StyledImage>;

export function Image(props: ImageProps) {
  return <StyledImage {...props} />;
}
