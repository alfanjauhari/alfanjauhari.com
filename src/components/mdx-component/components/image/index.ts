export * from './image';
export * from './image.styles';
