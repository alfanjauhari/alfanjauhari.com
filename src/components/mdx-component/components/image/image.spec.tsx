import { render, screen } from '@testing-library/react';
import { Image } from './image';

describe('MDXComponent/Image', () => {
  it('should render successfuly', () => {
    render(<Image alt="Title" src="/images/icon.webp" fill />);

    expect(screen.getByRole('img')).toBeVisible();
  });

  it('should match snapshot', () => {
    render(<Image alt="Title" src="/images/icon.webp" fill />);

    expect(screen.getByRole('img')).toMatchSnapshot();
  });
});
