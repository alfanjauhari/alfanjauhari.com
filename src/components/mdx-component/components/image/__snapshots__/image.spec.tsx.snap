// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXComponent/Image should match snapshot 1`] = `
<img
  alt="Title"
  class="c-juUgRT"
  data-nimg="fill"
  decoding="async"
  loading="lazy"
  sizes="100vw"
  src="/_next/image?url=%2Fimages%2Ficon.webp&w=3840&q=75"
  srcset="/_next/image?url=%2Fimages%2Ficon.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Ficon.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Ficon.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Ficon.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Ficon.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Ficon.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Ficon.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Ficon.webp&w=3840&q=75 3840w"
  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
/>
`;
